File: daily_challenge_leetcode/2483.js
/**
 * @param {string} customers
 * @return {number}
 */
var bestClosingTime = function (customers) {
    const n = customers.length
    let minHour = 0
    let penalty = [...customers].filter(c => c === 'Y').length
    let minPenalty = penalty
    for (let i = 0; i < n; i++) {
        if (customers[i] === 'Y') {
            penalty--
        } else {
            penalty++
        }
        if (penalty < minPenalty) {
            minHour = i + 1
            minPenalty = penalty
        }
    }
    return minHour
};

File: daily_challenge_leetcode/2483.py
class Solution:
    def bestClosingTime(self, customers: str) -> int:
        n = len(customers)
        minHour = 0
        penalty = customers.count("Y")
        minPenalty = penalty
        for i in range(n):
            if customers[i] == "Y":
                penalty -= 1
            else:
                penalty += 1
            if penalty < minPenalty:
                minHour = i + 1
                minPenalty = penalty
        return minHour
